commonv5/ts/sonarqube/Endpoint.ts

```diff
@@ -16,26 +18,58 @@ export interface EndpointData {
 }
 
 export default class Endpoint {
-  constructor(
-    public type: EndpointType,
-    private readonly data: EndpointData,
-  ) {}
+  public type: EndpointType;
 
-  public get auth() {
-    if (!this.data.token && this.data.password && this.data.password.length > 0) {
-      return { user: this.data.username, pass: this.data.password };
-    }
-    return { user: this.data.token || this.data.username };
-  }
+  private readonly data: EndpointData;
 
-  public get organization() {
-    return this.data.organization;
+  private readonly proxy: tl.ProxyConfiguration | null;
+
+  constructor(type: EndpointType, data: EndpointData) {
+    this.type = type;
+    this.data = data;
+    // Remove trailing slash at the end of the base url, if any
+    if (this.data) {
+      this.data.url = this.data?.url.replace(/\/$/, "");
+    }
+    this.proxy = tl.getHttpProxyConfiguration();
   }
 
   public get url() {
     return this.data.url;
   }
 
+  public get auth(): { username: string; password: string } {
+    // If using user/password
+    if (!this.data.token && this.data.password && this.data.password.length > 0) {
+      return { username: this.data.username, password: this.data.password };
+    }
+    // Using token
+    return { username: this.data.token || this.data.username, password: "" };
+  }
+
+  toFetch(): Partial<RequestInit> {
+    const options: Partial<RequestInit> = {
+      method: "get",
+    };
+
+    // Add HTTP auth from this.auth
+    options.headers = {
+      Authorization: `Basic ${Buffer.from(`${this.auth.username}:${this.auth.password}`).toString(
+        "base64",
+      )}`,
+    };
+
+    // Add proxy configuration, when relevant
+    if (this.proxy) {
+      options.agent = proxyAgent(this.proxy.proxyFormattedUrl);
+      tl.debug("Using proxy agent: " + JSON.stringify(options.agent));
+    } else {
+      tl.debug("Not using a proxy agent");
+    }
```

Comment:
Uses the agent proxy conf to set proxy for requests

Reply:
Agree. We have `toSonarProps` which is similar in the same class which generates a partial scanner property object. This method is very similar as it generates partial node-fetch options, what about `toFetchOptions` to have better meaning but still make it obvious it's different than `toSonarProps` semantically?